crates/scheduler/CHANGELOG.md

```diff
@@ -4,6 +4,7 @@
 
 ### Minor
 
+- Add `Failure` type
```

Comment:
This is not a minor change. This is purely an implementation detail, so a patch change. And the change should be something along the lines of "Simplify try-blocks for API call replies with a common `Failure` type".

Reply:
Moved to patch, but kept the description the same as I plan to do the other parts in following PRs. Please let me know if you would like me to do all of the steps in a single PR.

Reply:
Yes, I think it should all be in a single PR, since the changes independently don't make sense.

Reply:
Hey @ia0, I changed `Scheduler::reply()` to take the new `Result<_, Failure>` type as an argument and fixed all the compiler errors.